src/sorting.jl

```diff
@@ -73,7 +73,12 @@ Uses block y index to decide which values to operate on.
     sync_threads()
     blockIdx_yz = (blockIdx().z - 1i32) * gridDim().y + blockIdx().y
     idx0 = lo + (blockIdx_yz - 1i32) * blockDim().x + threadIdx().x
-    val = idx0 <= hi ? values[idx0] : one(eltype(values))
+    val = if idx0 <= hi
+        values[idx0]
+    else
+        Ref{eltype(values)}()[] # undef
+        # if idx0 > hi, val, comparison and dest_idx are unused
```

Comment:
@xaellison any thoughts on invoking `flex_lt` with such bogus values?

Reply:
Should be no problem since data is moved based on `sums`, and this doesn't affect that array